### PR TITLE
IIIF Viewer Refactor Tests phase F: Mobile, edge cases, and ViewerBottomBar

### PR DESCRIPTION
**Pull request by rcantin-w**
- For #12984 

## What does this change?

This PR completes comprehensive test coverage for the Item Viewer ahead of [RFC 086 IIIF Viewer Context Refactoring](https://github.com/wellcomecollection/docs/tree/main/rfcs/086-iiif-viewer-context). This is phase F of the testing work (following phases [A](https://github.com/wellcomecollection/wellcomecollection.org/pull/11531), [B](https://github.com/wellcomecollection/wellcomecollection.org/pull/11532), [C](https://github.com/wellcomecollection/wellcomecollection.org/pull/11533), and [D](https://github.com/wellcomecollection/wellcomecollection.org/pull/11534)).

This PR adds the missing test coverage identified during the RFC review:
- **Mobile-specific behaviour** (sidebar toggle, auto-close, control hiding)
- **Refactor-specific integration scenarios** (browser history, rapid navigation, deep linking, volume switching)
- **ViewerBottomBar component unit tests** (navigation, controls, edge cases)
- **ViewerTopBar edge case tests** (invalid canvas handling)

### E2E tests added (7 new tests in `view-items.test.ts`)

**Mobile-specific behaviour (3 tests):**
- Test 41: Mobile sidebar toggle text changes between "Show info" ↔ "Hide info"
- Test 42: Mobile sidebar closes automatically when navigating between canvases
- Test 43: Page and Grid view controls hide when mobile sidebar is open

**Refactor-specific integration scenarios (4 tests):**
- ~Test 44: Browser back/forward maintains correct canvas state~
- Test 44: Rapid canvas navigation doesn't cause state corruption
- Test 45: Direct URL to specific canvas loads correctly
- ~Test 47: Volume switching resets to first canvas~

These tests ensure the refactor maintains mobile-specific functionality and handles state management edge cases that could cause bugs during the context refactor.

### Component unit tests added

**New file: `ViewerBottomBar.test.tsx` (16 tests total):**
- Navigation button states (disabled at boundaries, enabled in middle)
- Page/Grid toggle visibility (hidden when sidebar active or zoomed)
- Fullscreen button visibility (requires isEnhanced && isFullscreenEnabled)
- Edge cases (invalid canvas numbers handled gracefully)

**Extended: `ViewerTopBar.test.tsx` (+2 edge case tests):**
- Handles canvas beyond array bounds (displays value without crashing)
- Handles canvas=0 (invalid 1-indexed value, displays without crashing)

### Helper improvements

**Updated `multiVolumeItem` in `contexts.ts`:**
- Now accepts optional `canvas` parameter for deep linking tests
- Usage: `multiVolumeItem(context, page)` or `multiVolumeItem(context, page, { canvasNumber: 5 })`
- Provides consistent test setup for canvas-specific scenarios

## How to test

All tests pass

## Have we considered potential risks?

**Low risk** - Only adds tests, no production code changes.

### What could go wrong?
- **E2E tests are flaky?** Mobile viewport tests use established `isMobile(page)` helper and wait patterns. Edge case tests use URL manipulation and standard assertions.
- **Missing coverage?** No - this completes the coverage gaps identified in the RFC review. All mobile behaviours, edge cases, and refactor-specific scenarios now tested.
- **Test duplication?** No - reviewed all existing tests to avoid duplication. Test 50 (rotation persistence) was removed as redundant with test 9.

No alarms needed - tests only.

### CODE REVIEW

File: playwright/test/view-items.test.ts [hunk 1]
```diff
@@ -601,3 +610,211 @@ test('(40) | OCR text should be accessible to screenreaders', async ({
...
+
+  // Navigate to canvas 2
+  const secondUrl = page.url().replace(/canvas=1/, 'canvas=2');
+  await page.goto(secondUrl);
+  await expect(page).toHaveURL(/canvas=2/);
+
+  // Navigate to canvas 3
+  const thirdUrl = page.url().replace(/canvas=2/, 'canvas=3');
+  await page.goto(thirdUrl);
+  await expect(page).toHaveURL(/canvas=3/);
```
[the remark this answers is not in the document]
Comment (rcantin-w) [reply playwright/test/view-items.test.ts]: I'll just pass in the canvas number, we've added that as a possibility with this helper.

File: playwright/test/helpers/contexts.ts [hunk 2]
```diff
@@ -90,10 +90,16 @@ export const requiredCookies = useStageApis
...
+  params?: { canvasNumber?: number }
 ): Promise<void> => {
   await context.addCookies(requiredCookies);
-  await gotoWithoutCache(`${baseUrl}/works/mg56yqa4/items`, page);
+  await gotoWithoutCache(
+    `${baseUrl}/works/mg56yqa4/items${
+      params?.canvasNumber ? `?canvas=${params.canvasNumber}` : ''
+    }`,
+    page
+  );
```
[the remark this answers is not in the document]
Comment (rcantin-w) [reply playwright/test/helpers/contexts.ts:102]: That's fine, we control the tests.

**Comment by rcantin-w** [comment]
Tests removed:

Test 44 (Browser back/forward): Removed due to flakiness. Browser history behaviour is unreliable in E2E tests because the navigation history isn't consistently set up. This functionality is already handled by Next.js routing (their responsibility) and we have sufficient coverage through tests that verify URL updates on navigation (tests 23, 35, 45).

Test 46 (Volume switching canvas reset): Removed due to flakiness and redundancy. The test attempted to verify whether switching volumes resets or maintains canvas position, but this is an implementation detail dependent on how volume links are constructed. We already test that volume navigation works (tests 10, 11) and that direct URLs to specific canvases work (test 45), which provides adequate coverage.

Both tests proved unreliable across test runs without providing coverage that wasn't already captured by more stable tests.

**Review by davidpmccormick — APPROVED** [review]
Makes sense. Tests pass
